source/gui/blockAction.py

```diff
@@ -0,0 +1,57 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import config
+from dataclasses import dataclass
+from enum import Enum
+from functools import wraps
+import globalVars
+from typing import Callable
+import ui
+from gui.message import isModalMessageBoxActive
+
+
+@dataclass
+class _Context:
+	blockActionIf: Callable[[], bool]
+	messageString: str
+
+
+class Context(_Context, Enum):
+	SECURE_MODE = (
+		lambda: globalVars.appArgs.secure,
+		# Translators: Reported when an action cannot be performed because NVDA is in a secure screen
+		_("Action unavailable in secure context"),
+	)
+	WINDOWS_STORE_VERSION = (
+		lambda: config.isAppX,
+		# Translators: Reported when an action cannot be performed because NVDA has been installed
+		# from the Windows Store.
+		_("Action unavailable in Windows Store version"),
+	)
+	MODAL_DIALOG_OPEN = (
+		isModalMessageBoxActive,
+		# Translators: Reported when an action cannot be performed because NVDA is waiting
+		# for a response from a modal dialog
+		_("Action unavailable while dialog response required"),

```

Comment:
Below a suggestion that seems clearer for the end user and also grammatically more complete rather than telegraphic style.
Telegraphic style (e.g. "Call me when train arrived" instead of "Call me when the train has arrived") may also be harder to translate for translators.


```suggestion
		_("Action unavailable since a response is required in another opened dialog."),
```

Feel free to rephrase it better if needed; English is not my native language.

Reply:
What do you think of the shorter:
```suggestion
		_("Action unavailable as an open dialog requires a response"),
```

Reply:
Yes very good. Your suggestion sounds much better ans shorter than mine.